xla/tools/compute_cost.cc

```diff
@@ -72,14 +77,39 @@ int main(int argc, char** argv) {
     analysis = std::make_unique<xla::HloCostAnalysis>();
   }
 
-  TF_CHECK_OK(xla::LoadModuleFromFile(input, format, {})
-                  .value()
-                  ->entry_computation()
-                  ->root_instruction()
-                  ->Accept(&*analysis));
+  std::unique_ptr<xla::HloModule> module =
+      *xla::LoadModuleFromFile(input, format, {});
+
+  TF_CHECK_OK(
+      module->entry_computation()->root_instruction()->Accept(&*analysis));
 
-  std::cout << std::setw(5) << std::setprecision(4)
-            << analysis->flop_count() / (1e9) << " GFLOPS. "
-            << analysis->bytes_accessed() / (1e6) << " MB." << std::endl;
+  if (all) {
+    absl::flat_hash_map<std::string, std::string> fingerprint_to_name;
```

Comment:
Please put the code from this branch in a separate function.

Reply:
done